tests/test_tensor_interface.py

```diff
@@ -0,0 +1,159 @@
+import pytest
+
+import aesara
+import aesara.sparse as sparse
+import aesara.tensor as at
+
+
+@pytest.mark.parametrize("is_sparse", [False, True])
+# @pytest.mark.parametrize("is_sparse", [False])
+class TestTensorMethods:
+    """checks that dense and sparse tensors have the same interface"""
+
+    @staticmethod
+    def _get_x(is_sparse):
+        x = at.dmatrix("x")
+        if is_sparse:
+            x = sparse.csr_from_dense(x)
+        return x
+
+    @staticmethod
+    def _get_xy(is_sparse):
+        x = at.lmatrix("x")
+        y = at.lmatrix("y")
+        if is_sparse:
+            x = sparse.csr_from_dense(x)
+            y = sparse.csr_from_dense(y)
+        return x, y
+
+    @pytest.mark.parametrize(
+        "method",
+        [
+            "__abs__",
+            "__neg__",
+            "__ceil__",
+            "__floor__",
+            "__trunc__",
+            "transpose",
+            "any",
+            "all",
+            "flatten",
+            "ravel",
+            "arccos",
+            "arcsin",
+            "arctan",
+            "arccosh",
+            "arcsinh",
+            "arctanh",
+            "ceil",
+            "cos",
+            "cosh",
+            "deg2rad",
+            "exp",
+            "exp2",
+            "expm1",
+            "floor",
+            "log",
+            "log10",
+            "log1p",
+            "log2",
+            "rad2deg",
+            "sin",
+            "sinh",
+            "sqrt",
+            "tan",
+            "tanh",
+            "copy",
+            "sum",
+            "prod",
+            "mean",
+            "var",
+            "std",
+            "min",
+            "max",
+            "argmin",
+            "argmax",
+            "nonzero",
+            "nonzero_values",
+            "argsort",
+            "conj",
+            "round",
+            "trace",
+            "zeros_like",
+            "ones_like",
+            "cumsum",
+            "cumprod",
+            "ptp",
+            "squeeze",
+            "diagonal",
+        ],
+    )
+    def test_unary(self, is_sparse, method):
+        x = self._get_x(is_sparse)
+        method_to_call = getattr(x, method)
+        z = method_to_call()
+        f = aesara.function([x], z, on_unused_input="ignore")
+        print(f([[1.1, 0.0, 2.0], [-1.0, 0.0, 0.0]]))
```

Comment:
We require that tests utilize explicit `assert`s.  This helps clarify the expectations of the function being tested and the tests themselves.